src/Foundatio.Aliyun/Storage/AliyunFileStorage.cs

```diff
@@ -0,0 +1,208 @@
+                    result => _client.EndCopyResult(result),
+                    new CopyObjectRequest(_bucketName, NormalizePath(path), _bucketName, NormalizePath(targetpath)), null);
+                return copyResult.HttpStatusCode == HttpStatusCode.OK;
+            }
+            catch (Exception) {
+                return false;
+            }
+        }
+
+        public Task<bool> DeleteFileAsync(string path, CancellationToken cancellationToken = default(CancellationToken)) {
+            if (string.IsNullOrEmpty(path)) {
+                throw new ArgumentNullException(nameof(path));
+            }
+            try {
+                _client.DeleteObject(_bucketName, NormalizePath(path));
+                return Task.FromResult(true);
+            }
+            catch (Exception) {
+                return Task.FromResult(false);
+            }
+        }
+
+        public async Task DeleteFilesAsync(string searchPattern = null, CancellationToken cancellation = default(CancellationToken)) {
+            var files = await GetFileListAsync(searchPattern, cancellationToken: cancellation).AnyContext();
+            _client.DeleteObjects(new DeleteObjectsRequest(_bucketName, files.Select(spec => spec.Path).ToList()));
+        }
+
+        public async Task<IEnumerable<FileSpec>> GetFileListAsync(string searchPattern = null, int? limit = null, int? skip = null,
+            CancellationToken cancellationToken = default(CancellationToken)) {
+            if (limit.HasValue && limit.Value <= 0)
+                return new List<FileSpec>();
+
+            searchPattern = searchPattern?.Replace('\\', '/');
+            string prefix = searchPattern;
+            Regex patternRegex = null;
+            int wildcardPos = searchPattern?.IndexOf('*') ?? -1;
+            if (searchPattern != null && wildcardPos >= 0) {
+                patternRegex = new Regex("^" + Regex.Escape(searchPattern).Replace("\\*", ".*?") + "$");
+                int slashPos = searchPattern.LastIndexOf('/');
+                prefix = slashPos >= 0 ? searchPattern.Substring(0, slashPos) : String.Empty;
+            }
+            prefix = prefix ?? String.Empty;
+
+            string marker = null;
+            var blobs = new List<OssObjectSummary>();
+            do {
+                var listing = await Task.Factory.FromAsync(
+                    (request, callback, state) => _client.BeginListObjects(request, callback, state),
+                    result => _client.EndListObjects(result), new ListObjectsRequest(_bucketName) {
+                        Prefix = prefix,
+                        Marker = marker,
+                        MaxKeys = limit
+                    }, null);
+                marker = listing.NextMarker;
+
+                blobs.AddRange(listing.ObjectSummaries.Where(blob => patternRegex == null || patternRegex.IsMatch(blob.Key)));
+            } while (!string.IsNullOrEmpty(marker) && blobs.Count < limit.GetValueOrDefault(Int32.MaxValue));

```

Comment:
We need to be checking the cancellation token otherwise this might never return (https://github.com/FoundatioFx/Foundatio.AzureStorage/blob/feature/logging/src/Foundatio.AzureStorage/Storage/AzureFileStorage.cs#L113)

Reply:
The CancellationToken type is struct, and never to be null. So I think we should to check cancellationToken.IsCancelRequested